embeddings/data/datamodule.py

```diff
@@ -71,10 +75,18 @@ def __init__(
         )
         self.load_dataset_kwargs = load_dataset_kwargs if load_dataset_kwargs else {}
 
-    def load_dataset(self) -> DatasetDict:
-        result = datasets.load_dataset(self.dataset_name, **self.load_dataset_kwargs)
-        assert isinstance(result, DatasetDict)
-        return result
+    def load_dataset(self, preparation_step: bool = False) -> DatasetDict:
+        loader: Union[HuggingFaceDataLoader, HuggingFaceLocalDataLoader] = HuggingFaceDataLoader()
+        if os.path.exists(self.dataset_name_or_path):
```

Comment:
```suggestion
        if exists(self.dataset_name_or_path):
```

and correct import accordingly